src/ImageSharp/Formats/Jpeg/Port/Components/FrameComponent.cs

```diff
@@ -0,0 +1,74 @@
+﻿// <copyright file="FrameComponent.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.Formats.Jpeg.Port.Components
+{
+    using System;
+
+    using ImageSharp.Memory;
+
+    /// <summary>
+    /// Represents a single frame component
+    /// </summary>
+    internal struct FrameComponent : IDisposable
+    {
+        /// <summary>
+        /// Gets or sets the component Id
+        /// </summary>
+        public byte Id;
+
+        /// <summary>
+        /// TODO: What does pred stand for?
+        /// </summary>
+        public int Pred;
+
+        /// <summary>
+        /// Gets or sets the horizontal sampling factor.
+        /// </summary>
+        public int HorizontalFactor;
+
+        /// <summary>
+        /// Gets or sets the vertical sampling factor.
+        /// </summary>
+        public int VerticalFactor;
+
+        /// <summary>
+        /// Gets or sets the identifier
+        /// </summary>
+        public byte QuantizationIdentifier;
+
+        /// <summary>
+        /// Gets or sets the block data
+        /// </summary>
+        public Buffer<short> BlockData;
```

Comment:
Okkay, this one seems easy for me ... but I really don't want to code all the refactors and tests by my own. You have better knowledge on the new decoder code now.
We also got the integration testing task as a jpeg optimization prerequisite. I want to focus on that in the next 1-2 weeks.